src/OrchardCore.Modules/OrchardCore.Resources/OrchardCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.ResourceManagement;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Resources;
+
+public static class OrchardCoreBuilderExtensions
+{
+    public static OrchardCoreBuilder ConfigureResourceSettings(this OrchardCoreBuilder builder)
+    {
+        builder.ConfigureServices((tenantServices, serviceProvider) =>
+        {
+            var configurationSection = serviceProvider.GetRequiredService<IShellConfiguration>().GetSection("OrchardCore_Resources");
+
+            tenantServices.PostConfigure<ResourceOptions>(settings =>
+            {
+                configurationSection.Bind(settings);
+
+                settings.ResourceDebugMode = ResourceDebugMode.FromConfiguration;
```

Comment:
Why not also binding this property, maybe with a default value.

Don't change it but I think the right enum name would be `ResourceDebugMode.FromEnvironmentName`.

When we talked about `.FromConfiguration` for `SecuritySettings` it was not the same, it is a dedicated property to say that all options come from config, it is set to true in `ConfigureSecuritySettings()`, then it is used in `SecuritySettingsDisplayDriver` and in the `SecurityHeadersSettings.Edit.cshtml` view.

The driver injects `IOptionsSnapshot<SecuritySettings>`, and if `.FromConfiguration` is true the view model is not populated from the persisted settings but from the config values, so that when editing the settings we see the winning settings that are currently used. And the view says `The current settings are coming from configuration sources ...`. Not sure I like this pattern but this is what is currently done.